### PR TITLE
ENH unittests for naive bayes

### PR DESCRIPTION
**Pull request by neonrights**
Created unittests for naive bayes methods. Currently tests constructors and methods using a univariate distribution, bivariate distribution, and hidden markov models.

### CODE REVIEW

File: tests/test_naive_bayes.py [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+	assert_almost_equal( normal.log_probability( 5 ), -1.612085713764219 )
+	assert_almost_equal( normal.log_probability( 3 ), -2.112085713764219 )
+	assert_almost_equal( normal.log_probability( 1 ), -3.612085713764219 )
+	assert_almost_equal( normal.log_probability( -1 ), -6.112085713764219 )
+
+	assert_almost_equal( uniform.log_probability( 11 ), -float('inf') )
+	assert_almost_equal( uniform.log_probability( 10 ), -2.3025850929940455 )
+	assert_almost_equal( uniform.log_probability( 5 ), -2.3025850929940455 )
+	assert_almost_equal( uniform.log_probability( 0 ), -2.3025850929940455 )
+	assert_almost_equal( uniform.log_probability( -1 ), -float('inf') )
```
Comment (jmschrei) [inline tests/test_naive_bayes.py]: Good unit tests, but put these in their own function, not the setup_models one

File: tests/test_naive_bayes.py [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+	assert_almost_equal( multi.log_probability([ 5, 5 ]), -2.5310242469699453 )
+	assert_almost_equal( multi.log_probability([ 3, 3 ]), -4.531024246969945 )
+	assert_almost_equal( multi.log_probability([ 1, 1 ]), -10.531024246969945 )
+	assert_almost_equal( multi.log_probability([ -1, -1 ]), -20.531024246969945 )
+
+	assert_almost_equal( indie.log_probability([ 11, 11 ]), -float('inf') )
+	assert_almost_equal( indie.log_probability([ 10, 10 ]), -4.605170185988091 )
+	assert_almost_equal( indie.log_probability([ 5, 5 ]), -4.605170185988091 )
+	assert_almost_equal( indie.log_probability([ 0, 0 ]), -4.605170185988091 )
+	assert_almost_equal( indie.log_probability([ -1, -1 ]), -float('inf') )
```
Comment (jmschrei) [inline tests/test_naive_bayes.py]: put these in their own function too. test_univariate_nb and test_multivariate_nb or something

File: tests/test_naive_bayes.py [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+	smart = HiddenMarkovModel()
+	smart.add_transition( smart.start, unrigged, 0.5 )
+	smart.add_transition( smart.start, rigged, 0.5 )
+	smart.add_transition( rigged, rigged, 0.5 )
+	smart.add_transition( rigged, unrigged, 0.5 )
+	smart.add_transition( unrigged, rigged, 0.5 )
+	smart.add_transition( unrigged, unrigged, 0.5 )
+	smart.bake()
+
+	# make sure hmm's are working
```
Comment (jmschrei) [inline tests/test_naive_bayes.py]: basically the setup function just produces the models. no testing should happen in them. the testing happens in their own functions later on

File: tests/test_naive_bayes.py [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+	assert_almost_equal( smart.log_probability( list('TTTT') ), -4.1992884979947105 )
+	assert_almost_equal( smart.log_probability( list('THTHTHTHTHTH') ), -8.883630243546788 )
+	assert_almost_equal( smart.log_probability( list('THTHHHHHTHTH') ), -7.645551826734343 )
+
+	# to be used in each test
+	global univariate
+	global multivariate
+	global hmms
+
+	# no errors should be thrown
```
Comment (jmschrei) [inline tests/test_naive_bayes.py]: have one setup for each of the naive bayes models you're going to use, and then several tests for each model

**Comment by jmschrei** [comment]
How is this coming?

**Comment by neonrights** [comment]
I've broken up the different naive bayes and distributions into their separate test methods. The basic tests should be done now. I think I pushed those changes last thursday or wednesday.

**Comment by jmschrei** [comment]
Awesome. I reran the failing unit tests and it looks like they're working now.

**Pull request merged**